pkg/kubectl/rolling_updater.go

```diff
@@ -208,9 +208,15 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 	if err != nil {
 		return err
 	}
-	// Further validation.
-	if maxUnavailable == 0 && maxSurge == 0 {
-		return fmt.Errorf("one of maxSurge or maxUnavailable must be specified")
+	// If the desired new scale is 0, then the max unavailable is necessarily
+	// the effective scale of the old RC regardless of the configuration
+	// (equivalent to 100% maxUnavailable).
+	if desired == 0 {
+		maxUnavailable = int(math.Max(float64(oldRc.Spec.Replicas), float64(oldRc.Status.Replicas)))
```

Comment:
What we really want is minAvailable to be zero on line 233, so I think setting this to original is better.
We can actually set minAvailable directly (thats the only use of maxUnavailable)